src/Paprika/Data/SlottedArray.cs

```diff
@@ -108,7 +108,11 @@ private bool TrySetImpl(ushort hash, byte preamble, in NibblePath trimmed, ReadO
         // write item: length_key, key, data
         var dest = _data.Slice(slot.ItemAddress, total);
 
-        if (HasKeyBytes(preamble))
+        if (HasLengthFourToSix(preamble))
```

Comment:
I'd consider running benchmarks against this, and then change the branching structure to limit the checks. Right now, if the path is longer than 6, you'll perform two checks. One could change it to something

```
if (length <= 6)
{
  if (length < 4)
  ...
  else
}
else
```

Again, benchmarking needed.

Reply:
I'm still trying to implement the functionality beforehand, and will focus on the optimizations as soon as this starts working 🫡🫡